src/test/java/UserServiceTests.java

```diff
@@ -0,0 +1,173 @@
+
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.web.client.RestTemplate;
+import recipeapplication.RecipeApplication;
+import recipeapplication.boundaries.UserEditTO;
+import recipeapplication.components.Name;
+import recipeapplication.components.Role;
+import recipeapplication.components.User;
+import recipeapplication.exceptions.UserNotFoundException;
+import recipeapplication.services.UserService;
+
+import java.util.ArrayList;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+
+@Component
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = RecipeApplication.class, webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@TestPropertySource(properties = "server.port=8080")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class UserServiceTests {
+
+
+    private int port;
+    private String baseUrl;
+    private RestTemplate restTemplate;
+    private static String id;
+    private static int count = 0;
+    private boolean deleteTest = false;
+
+    @Autowired
+    private UserService userService;
+
+    public UserServiceTests() {
+
+    }
+    @LocalServerPort
+    public void setPort(int port) {
+        this.port = port;
+    }
+
+    @BeforeAll
+    public void init() {
+        this.baseUrl = "http://localhost:" + port + "/user";
+        this.restTemplate = new RestTemplate();
+    }
+
+    @AfterEach
+    public void tearDown() {
+        if (!deleteTest)
+            this.userService.deleteUser(id);
+    }
+
+    @Test
+    public void addNewUserTest() {
+        deleteTest = false;
+        // GIVEN the database is clean
+
+        // WHEN I post a new user
+        User newUser = new User();
+        fillDetailsNewUser(newUser);
+        User response = this.restTemplate
+                .postForObject(
+                        this.baseUrl,
+                        newUser,
+                        User.class);
+
+        // THEN the database contains 1 user
+        // AND the returned user is similar to the user in the database
+        assertThat(response)
+                .isNotNull();
+        assertThat(this.userService.readUser(id))
```

Comment:
Very inefficient way of doing this,
read the user once from the db and then make assets on it